weaviate/collection/classes.py

```diff
@@ -266,27 +266,19 @@ class _MetadataReturn:
     explain_score: Optional[str] = None
     is_consistent: Optional[bool] = None
 
-    def __init__(self, data: Optional[Dict[str, Any]] = None) -> None:
-        if data is None:
-            return
-
-        def _to_uuid(uuid_str: Optional[str]) -> Optional[uuid_package.UUID]:
-            if uuid_str is None:
-                return None
-            return uuid_package.UUID(hex=uuid_str)
-
-        def _parse(key: str) -> Optional[Any]:
-            return data.get(key)
-
-        self.uuid = _to_uuid(_parse("id"))
-        self.vector = _parse("vector")
-        self.creation_time_unix = _parse("creationTimeUnix")
-        self.last_update_time_unix = _parse("lastUpdateTimeUnix")
-        self.distance = _parse("distance")
-        self.certainty = _parse("certainty")
-        self.score = _parse("score")
-        self.explain_score = _parse("explainScore")
-        self.is_consistent = _parse("isConsistent")
+
+def _MetadataFromDict(metadata: Dict[str, Any]) -> _MetadataReturn:
+    return _MetadataReturn(
+        uuid=uuid_package.UUID(metadata["id"]) if "id" in metadata else None,
+        vector=metadata.get("vector"),
+        creation_time_unix=metadata.get("creationTimeUnix"),
+        last_update_time_unix=metadata.get("lastUpdateTimeUnix"),
+        distance=metadata.get("distance"),
+        certainty=metadata.get("certainty"),
+        explain_score=metadata.get("explainScore"),
+        score=metadata.get("score"),
+        is_consistent=metadata.get("isConsistent"),
+    )
```

Comment:
yeah, I thought about it.

I don't feel strongly about this and agree that your solution is nicer but I'd like to avoid to clutter the functions that the users get with things they are not supposed to use.

Reply:
Sadly, another instance of improving userX at the expense of devX, alas!

Reply:
Can the name of the function be changed to make its appearance as pythonic as possible? So something like:
```python
def _metadata_from_dict()
```

Reply:
sure